tc_aws/loaders/s3_loader.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, key, callback=None,
         self.bucket_loader = bucket_loader
         self.callback = callback
         self.context = context
-        self.limit_max_retries = context.config.get('TC_AWS_MAX_RETRIES')
+        self.limit_max_retries = context.config.get('TC_AWS_MAX_RETRY')
```

Comment:
Maybe we should rename the property as well to avoid confusions?

Reply:
What do you think if caller passes `max_retry` as argument instead of passing whole context object? I suspect (no proofs, though), that it could create another unneeded circular reference.

Reply:
Yes, seems good as context is not needed elsewhere :)